homeassistant/components/lektrico/sensor.py

```diff
@@ -0,0 +1,206 @@
+        value_fn=lambda x: x.voltage,
+    ),
+    LektricoSensorEntityDescription(
+        key="install_current",
+        name="Install Current",
+        device_class=SensorDeviceClass.CURRENT,
+        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+        value_fn=lambda x: x.install_current,
+    ),
+    LektricoSensorEntityDescription(
+        key="dynamic_current",
+        name="Dynamic Current",
+        device_class=SensorDeviceClass.CURRENT,
+        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+        value_fn=lambda x: x.dynamic_current,
+    ),
+    LektricoSensorEntityDescription(
+        key="led_max_brightness",
+        name="Led Brightness",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.ILLUMINANCE,
+        native_unit_of_measurement=PERCENTAGE,
+        value_fn=lambda x: x.led_max_brightness,
+    ),
+    LektricoSensorEntityDescription(
+        key="headless",
+        name="No Authentication",
+        value_fn=lambda x: x.headless,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Lektrico charger based on a config entry."""
+    _lektrico_device: LektricoDevice = hass.data[DOMAIN][entry.entry_id]
+
+    sensors = [
+        LektricoSensor(
+            sensor_desc,
+            _lektrico_device.serial_number,
+            _lektrico_device.board_revision,
+            _lektrico_device,
+            entry.data[CONF_FRIENDLY_NAME],
+        )
+        for sensor_desc in SENSORS
+    ]
+
+    async_add_entities(sensors, True)
+
+
+class LektricoSensor(CoordinatorEntity, SensorEntity):
+    """The entity class for Lektrico charging stations sensors."""
+
+    entity_description: LektricoSensorEntityDescription
+
+    def __init__(
+        self,
+        description: LektricoSensorEntityDescription,
+        serial_number: str,
+        board_revision: str,
+        _lektrico_device: LektricoDevice,
+        friendly_name: str,
+    ) -> None:
+        """Initialize Lektrico charger."""
+        super().__init__(_lektrico_device.coordinator)
+        self.friendly_name = friendly_name
+        self.serial_number = serial_number
+        self.board_revision = board_revision
+        self.entity_description = description
+
+        self._attr_name = f"{self.friendly_name} {description.name}"
+        self._attr_unique_id = f"{serial_number}_{description.name}"
+        # ex: 500006_Led Brightness
+
+        self._lektrico_device = _lektrico_device
+
+    @property
+    def native_value(self) -> float | str | Any | None:
+        """Return the state of the sensor."""
+        if self.entity_description.value_fn is not None:
+            return self.entity_description.value_fn(
+                self._lektrico_device.coordinator.data
+            )
+        return None
+
+    @property
+    def native_unit_of_measurement(self) -> str | None:
```

Comment:
This method can be completely deleted. `SensorEntity` does all of this code for you.

Reply:
I deleted it. Thank you.